listenbrainz_spark/utils.py

```diff
@@ -239,3 +235,21 @@ def path_exists(path):
     if path_found:
         return True
     return False
+
+def hdfs_walk(path, depth=0):
+    """ Depth-first walk of HDFS filesystem.
+
+        Args:
+            path (str): Path to start DFS.
+            depth (int): Maximum depth to explore files/folders. 0 for no limit.
+
+        Returns:
+            walk: a generator yeilding tuples (path, dirs, files).
+    """
+    try:
+        walk = hdfs_connection.client.walk(hdfs_path=path, depth=depth)
+        return walk
+    except HdfsError as err:
+        raise PathNotFoundException(str(err), path)
+
```

Comment:
Extra whitespace, sorry to mention this everytime, but you should look into setting up your editor to remove these automatically

Reply:
I have changed the setting of my editor long ago, It sometimes behaves weirdly :(
I will look into it :)